chains/ethereum/writer.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package ethereum
+
+import (
+	"math/big"
+
+	"github.com/ChainSafe/ChainBridge/chains"
+	msg "github.com/ChainSafe/ChainBridge/message"
+	"github.com/ChainSafe/log15"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var _ chains.Writer = &Writer{}
+
+type Writer struct {
+	cfg                  Config
+	conn                 *Connection
+	bridgeContract       *BridgeContract // instance of bound receiver bridgeContract
+	erc20HandlerContract *ERC20HandlerContract
+	gasPrice             *big.Int
+	gasLimit             *big.Int
+	log                  log15.Logger
+}
+
+func NewWriter(conn *Connection, cfg *Config, log log15.Logger) *Writer {
+	return &Writer{
+		cfg:      *cfg,
+		conn:     conn,
+		gasPrice: cfg.gasPrice,
+		gasLimit: cfg.gasLimit,
+		log:      log,
+	}
+}
+
+func (w *Writer) Start() error {
+	w.log.Debug("Starting ethereum writer...")
+	return nil
+}
+
+func (w *Writer) SetContracts(bridge *BridgeContract, erc20Handler *ERC20HandlerContract) {
+	w.bridgeContract = bridge
+	w.erc20HandlerContract = erc20Handler
+}
+
+// ResolveMessage handles any given message based on type
+// A bool is returned to indicate failure/success, this should be ignored except for within tests.
+func (w *Writer) ResolveMessage(m msg.Message) bool {
+	w.log.Trace("Attempting to resolve message", "type", m.Type, "src", m.Source, "dst", m.Destination)
+
+	switch m.Type {
+	case msg.FungibleTransfer:
+		return w.createErc20DepositProposal(m)
+	case msg.NonFungibleTransfer:
+		panic("not yet implemented")
+	case msg.GenericTransfer:
+		return w.createGenericDepositProposal(m)
```

Comment:
I dont see the contract being set, I'm confused at the pattern being followed for handling generic transfers?

Reply:
All writes are to the bridge contract, updated in #320